File: Programmers/JS/DFSBFS_타겟넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    function BFS(idx, number, numbers, target) {
    if (idx === numbers.length){
        if (number === target) {
             answer += 1
        }
    } else {
        var i = idx + 1
        var plusNum = number + numbers[idx]
        var minusNum = number - numbers[idx]
        BFS(i, plusNum, numbers, target)
        BFS(i, minusNum, numbers, target)
    }
}
    BFS(0, 0, numbers, target)
    return answer;
}

File: Programmers/Python/DFSBFS_타겟넘버.py
def solve(lst, idx, num, tg_num):
    global answer
    if idx == len(lst):
        if num == tg_num:
            answer += 1
    else:
        solve(lst, idx+1, num + lst[idx], tg_num)
        solve(lst, idx+1, num - lst[idx], tg_num)

def solution(numbers, target):
    global answer
    answer = 0
    solve(numbers, 0, 0, target)
    return answer